libpod/rootless_cni_linux.go

```diff
@@ -308,14 +343,39 @@ func startRootlessCNIInfraContainer(ctx context.Context, r *Runtime) (*Container
 	return c, nil
 }
 
-func rootlessCNIInfraExec(c *Container, args ...string) (string, error) {
-	cmd := "rootless-cni-infra"
+func mountRootlessCNIINfraRootfs(rootfs string) error {
+	if err := os.MkdirAll(rootfs, 0700); err != nil {
+		return err
+	}
+	// bind mount the rootfs in the userns read only
+	if err := mount.Mount("/", rootfs, "bind", "rbind,rprivate,ro"); err != nil {
```

Comment:
It should be noted in comment lines that "rbind,ro" is just a "best effort" to make the tree read-only, as "ro" is not recursive.

Reply:
Good point. Is it possible to mount everything as readonly?

Reply:
Probably possible with `fuse-overlayfs`. We could also parse `/proc/self/mountinfo` and bind-mount each of the entries with `ro`, but that seems too complicated.

Reply:
Kernel patch for recursive read-only is here (unmerged): https://lore.kernel.org/linux-fsdevel/20210112220124.837960-7-christian.brauner@ubuntu.com/T/#u

Reply:
Would be great to get that in.